ndc-models/src/lib.rs

```diff
@@ -573,6 +573,16 @@ pub enum UnaryComparisonOperator {
 }
 // ANCHOR_END: UnaryComparisonOperator
 
+// ANCHOR: ColumnSelector
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, JsonSchema)]
+#[serde(untagged)]
```

Comment:
There's a couple of things I'd like to change here:

- We don't use `untagged` generally, so I'd like to avoid it.
- The `Path` String vector consists of two different types of string (names of fields, and names of JSON fields), so I'd like to change that.

So here is what I would suggest:

- Everywhere `ColumnSelector` is used, add an adjacent field of type `Option<Vec<String>>`, meaning the list of JSON field names to traverse.
- Remove `ColumnSelector`.

So the idea is we always have a _column_ name, and we _might_ have a list of JSON field names to traverse.

The capability would enable that optional field.

Reply:
Is there a meaningful difference between `Option<Vec<String>>` and `Vec<String>`? Not sure why we prefer the former over the latter.

Also, I would suggest having `Vec<String>` be a newtype `struct Path(Vec<String>)`.

Reply:
The idea was that we want to implement this as a non-breaking change, but we might be able to do that with `skip_serializing_if`.